lib/api/funnel.js

```diff
@@ -241,64 +246,65 @@ class Funnel {
       return 1;
     }
 
-    const processNow = this.getRequestSlot('execute', request, callback);
+    try {
+      const executing = this.throttle(req => {
+        if (req.error) {
+          // "handleErrorDump" shouldn't need to be called for 503 errors
+          callback(req.error, req);
+          return;
+        }
```

Comment:
This test is now obsolete: before it was outside the executor callback. In your new version of the code, this case is handled by the try..catch statement.

Furthermore, the callback is invoked by `throttle`, and it never invokes it if there is an error, thus this test is always false.